### PR TITLE
Start WebSocket server to broadcast auction real-time info, Improve logging

### PR DESCRIPTION
**Pull request by errfrom**

### CODE REVIEW

File: app/delegate-server/Lib/AVar.purs [hunk 1]
```diff
@@ -5,17 +5,17 @@ module DelegateServer.Lib.AVar
...
 import Effect.Aff.AVar (put, take) as AVar
+import Effect.Aff.Class (class MonadAff, liftAff)
 
-modifyAVar :: forall (a :: Type). AVar a -> (a -> Aff a) -> Aff a
+modifyAVar :: forall m a. MonadAff m => AVar a -> (a -> m a) -> m a
 modifyAVar avar f = do
-  val <- f =<< AVar.take avar
-  AVar.put val avar
+  val <- f =<< liftAff (AVar.take avar)
+  liftAff $ AVar.put val avar
```
Comment (euonymos) [inline app/delegate-server/Lib/AVar.purs:15]: Can it block here or not?

File: app/delegate-server/Types/AppExitReason.purs [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
 import Prelude
 
 import Data.Generic.Rep (class Generic)
 import Data.Show.Generic (genericShow)
+import DelegateServer.Types.HydraHeadStatus (HydraHeadStatus)
 
-data AppExitReason = AppExitReason_HeadFinalized
+data AppExitReason
+  = AppExitReason_HeadFinalized
+  | AppExitReason_BiddingTimeExpired_UnexpectedHeadStatus HydraHeadStatus
```
Comment (euonymos) [inline app/delegate-server/Types/AppExitReason.purs]: Is it one or two different reasons?

**Review by euonymos — APPROVED** [review]

File: app/delegate-server/Types/AppExitReason.purs (hunk 2 above)
Comment (errfrom) [reply app/delegate-server/Types/AppExitReason.purs]: It's two different reasons:

1. `AppExitReason_HeadFinalized`: Standing bid has been moved back to L1, final state has been reached, nothing left to be done.
2. `AppExitReason_BiddingTimeExpired_UnexpectedHeadStatus`: Bidding time expired, but the head has an unexpected status, that does not imply any further actions. It's actually only possible for `Idle` status, so I will update this constructor to `AppExitReason_BiddingTimeExpired_HeadIdle`.

Comment (errfrom) [reply app/delegate-server/Types/AppExitReason.purs]: Updated `AppExitReason` in https://github.com/mlabs-haskell/hydra-auction-offchain/pull/35/commits/67a4a268b775a6613b177046aec4a49bc707c847

File: app/delegate-server/Lib/AVar.purs (hunk 1 above)
Comment (errfrom) [reply app/delegate-server/Lib/AVar.purs:15]: These functions are not deadlock-safe and should be used with caution, namely, they should only be applied to AVars that are known to be non-empty or that are expected to be filled.

**Pull request merged**